snapcraft/providers/_provider.py

```diff
@@ -74,8 +74,17 @@ def ensure_provider_is_available(cls) -> None:
         """
 
     @staticmethod
-    def get_command_environment() -> Dict[str, Optional[str]]:
-        """Construct the required environment."""
+    def get_command_environment(
+        http_proxy: Optional[str] = None,
+        https_proxy: Optional[str] = None,
```

Comment:
we should add no_proxy too

Reply:
From our discussion:
1. If the environmental variable `no_proxy` is defined on the host system, it gets passed to the instance [here](https://github.com/snapcore/snapcraft/blob/7cc2a529e1e694b5ba72af5e516dac0ec6c793fd/snapcraft/providers/_provider.py#L86).  I tested and confirmed this.
2. `--no-proxy` is not a command line option, which is why I didn't add extra handling for it in this PR